packages/machine-extractor/src/types.ts

```diff
@@ -32,7 +32,7 @@ export interface Comment {
 }
 
 export interface FileExtractResult {
-  machines: (MachineExtractResult | undefined)[];
+  machines: MachineExtractResult[];
```

Comment:
I don't think this is a correct change. This `FileExtractResult["machines"]` type is also used [here](https://github.com/statelyai/xstate-tools/blob/f99f685e1f9d85749c7813e4b26133b6c2b2c4d0/packages/machine-extractor/src/extractMachinesFromFile.ts#L18) and I don't think that `MachineExtractResult` is guaranteed there. I actually don't like this and it would be cool to refactor this in the future but that would be a bigger refactor